### ISSUE TITLE
Content jumping around while loading tracks

### ISSUE THREAD

**Issue by iHiD** [exercism/website#175]
When searching/filtering etc, there's a lot of flashing happening caused by page elements jumping around:

https://www.loom.com/share/026b7c0909c84fd7a62e9fbf22547138

It feels like the whole page is being rerendered each time, causing everything to jump around, rather than just the parts that actually need to be re-rendered.

If only the relevant bits of content were changing the the page wouldn't be jumping and we wouldn't get this flashing effect.

We should fix this before doing more component stuff.

(cc @SleeplessByte for advice 🙂)

**Comment by SleeplessByte** [comment]
My tip for today:

Install React Dev Tools chrome plugin, and on the components tab, press settings, and turn on "highlight on updates".

**Comment by iHiD** [comment]
tl;dr EVERYTHING RE-RENDERS 😉
